### DISCUSSION TITLE
Buildah intermediate container(Id) is not resolvable

### DISCUSSION THREAD

**Discussion by SaurabhAhuja1983** [containers/buildah#4424] [Q&A]
### Docker  is assigning ipaddress to each intermediate container and containerId is resolvable. But buildah intermediate containerId is not resolvable...

```
I used following Dockerfile
FROM <REPO-TAKE-ANY-JAVA-IMAGE>/java_jdk_8:latest
COPY MyClass.java .

RUN echo "---" ; cat /etc/hosts ; echo "Hostname: $(hostname)" ; echo "---"; javac MyClass.java ; /usr/bin/java -cp . MyClass
```

```
cat MyClass.java

import java.net.InetAddress;
import java.net.UnknownHostException;

public class MyClass {
    public static void main(String args[]) {
        try {
            InetAddress ia = InetAddress.getLocalHost();
            String str = ia.getHostAddress();
            String HOSTNAME = InetAddress.getLocalHost().getHostName();
            System.out.println("hostname: " + HOSTNAME);
         } catch (final UnknownHostException e) {
            throw new IllegalStateException("cannot get host name", e);
         }
    }
}

```

```
See the following output
`buildah build --no-cache .
STEP 1/4: FROM <.............................>/java_jdk_8:latest
STEP 2/4: RUN which java
/usr/bin/java
STEP 3/4: COPY MyClass.java .
STEP 4/4: RUN echo "---" ; cat /etc/hosts ; echo "Hostname: $(hostname)" ; echo "---"; javac MyClass.java ; /usr/bin/java -cp . MyClass
---
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
fe00::0	ip6-mcastprefix
fe00::1	ip6-allnodes
fe00::2	ip6-allrouters
10.20.203.214	agent-79ccf9cbc-cvstd
10.20.203.214	host.containers.internal
Hostname: 43d4f3951829
---
Exception in thread "main" java.lang.IllegalStateException: cannot get host name
	at MyClass.main(MyClass.java:12)
Caused by: java.net.UnknownHostException: 43d4f3951829: 43d4f3951829: Name or service not known
	at java.net.InetAddress.getLocalHost(InetAddress.java:1418)
	at MyClass.main(MyClass.java:7)
Caused by: java.net.UnknownHostException: 43d4f3951829: Name or service not known
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:866)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1288)
	at java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:815)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
	at java.net.InetAddress.getLocalHost(InetAddress.java:1413)
	... 1 more
subprocess exited with status 1
subprocess exited with status 1
error building at STEP "RUN echo "---" ; cat /etc/hosts ; echo "Hostname: $(hostname)" ; echo "---"; javac MyClass.java ; /usr/bin/java -cp . MyClass": exit status 1
`
```

However the same code works fine with docker build as i could see the intermediate containerId to IP mapping in /etc/hosts.

Not sure if buildah intermediate container is not assigned an ipaddress but it's blocking our java projects from getting build as we are transitioning from docker to buildah for building images.

**Comment by SaurabhAhuja1983** [comment]
@rhatdan what's the right platform to discuss this problem?
I see from buildah's code that it makes ip -> host.containers.internal entry in /etc/hosts
but not ip -> containerId
Can we patch buildah to make this entry as well?
or can we specify hostname as  host.containers.internal to all the intermediate containers created while executing Containerfile?

It's kind of annoying that java libraries e.g. log4j calls getHostName() and fails with UnknownHostException.. which practically means we cannot build java apps using buildah if this bug is not resolved.

**Reply by rhatdan** [reply]
Yes buildah is the right place, open an issue to make it happen. Or even better a PR.

**Comment by SaurabhAhuja1983** [comment]
Thank You @rhatdan for opening the issue
https://github.com/containers/buildah/issues/4446

I was going through the code, Let me try to work on it and publish the PR.
let me go through the PR guidelines etc.
Also, Will be happy to work with someone who is already picking it up.

**Comment by SaurabhAhuja1983** [comment]
This is done as part of this PR:https://github.com/containers/buildah/pull/4869
Thank you.
